### ISSUE TITLE
Decoupling view and world logic.

### ISSUE THREAD

**Issue by TristanTrim** [traverseda/PyCraft#70]
I'm decoupling the view (drawing to canvas) from the world logic, but I'm finding that I need to either hand a 'world reference' to view, and a 'view reference' to world in main OR call functions on both of them from window, and hand data back and forth.

I'm after clean modularity, and down the line the possibility of running the world without a view. For that I think functions called by window would be better. But they both seem kinda kludgy.

Are there any pros and cons I'm missing?

**Comment by mrpudn** [comment]
When I split up the original `main.py`, I tried to move all of the opengl stuff into `World`, but I noticed some weird rendering errors when I did that. I'd say `World` shouldn't be much more than a glorified `dict`. `Window` can handle all of the opengl stuff and other rendering activities until we can break it down.

Thoughts?

**Comment by TristanTrim** [comment]
Yeah, I noticed there is opengl stuff in both window and world. Right now window is handling the update loop, user input, and some rendering stuff. I'd like to have window basically act as a dispatcher: calling update functions, and handling input from the user. It would call to update the view, but not handle all of the view logic itself.
